src/bindings/c/include/openvino/c/ov_core.h

```diff
@@ -189,6 +189,28 @@ ov_core_read_model_from_memory(const ov_core_t* core,
                                const ov_tensor_t* weights,
                                ov_model_t** model);
 
+/**
+ * @brief Reads models from IR / ONNX / PDPD / TF / TFLite formats with models string size.
+ * @ingroup ov_core_c_api
+ * @param core A pointer to the ie_core_t instance.
+ * @param model_str String with a model in IR / ONNX / PDPD / TF / TFLite format, support model string containing
+ * several null chars.
+ * @param str_len The length of model string.
+ * @param weights Shared pointer to a constant tensor with weights.
+ * @param model A pointer to the newly created model.
+ * Reading ONNX / PDPD / TF / TFLite models does not support loading weights from the @p weights tensors.
+ * @note Created model object shares the weights with the @p weights object.
+ * Thus, do not create @p weights on temporary data that can be freed later, since the model
+ * constant data will point to an invalid memory.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_core_read_model_from_memory_with_size(const ov_core_t* core,
```

Comment:
Maybe we can come up the better name... For example: `ov_core_read_model_from_memory_buffer`

Reply:
Also we can deprecate old method, because these methods provide the same functionality

Reply:
But if deprecate old method maybe lead to some compatibility problem if this API has been used by user application.

Reply:
Yes in this case user will get a deprecation warning, but it is a normal process and without it we cannot change or remove legacy API. I don't talk about deletion of legacy method immediately, only about the deprecation and removing later.
From my point of view two methods will confuse users...